packages/xo-web/src/common/xo/migrate-vm-modal/index.js

```diff
@@ -80,25 +88,20 @@ export default class MigrateVmModalBody extends BaseComponent {
       host => (host ? sr => isSrWritable(sr) && (sr.$container === host.id || sr.$container === host.$pool) : false)
     )
 
-    this._getTargetNetworkPredicate = createSelector(
-      createPicker(
-        () => this.props.pifs,
-        () => this.state.host.$PIFs
-      ),
-      pifs => {
-        if (!pifs) {
-          return false
-        }
-
-        const networks = {}
-        forEach(pifs, pif => {
-          networks[pif.$network] = true
-        })
-
-        return network => networks[network.id]
-      }
+    this._getNetworks = createCollectionWrapper(
+      createSelector(
+        () => this.props.networks,
+        () => this.state.host.$poolId,
+        (networks, poolId) => filter(networks, network => network.$poolId === poolId)
+      )
     )
 
+    this._getTargetNetworkPredicate = createSelector(this._getNetworks, networks => {
+      const _networks = {}
+      forEach(networks, network => (_networks[network.id] = true))
```

Comment:
What's the point of building an array of network and then build an object from that filtered array? Why don't you build the object only?